### DISCUSSION TITLE
Switching from echo to fiber?

### DISCUSSION THREAD

**Discussion by Nevoral** [pocketbase/pocketbase#2646] [Q&A] [answered]
Hi,
Why did you use echo instead of fiber? Is it because echo have more middleware? Could be possible to replace fiber in places where will be much bigger traffic or it would not be significant change in performance so it is not worth it?

Thanks for answer
Tomas

**Comment by ganigeorgiev** [comment, marked as answer]
I use echo because I like it, its code is fairly small and easy to understand.

At least for now there are no plans to switch to a different router. Eventually after the [std `http.ServeMux` enhancements get approved](https://github.com/golang/go/discussions/60227) I may consider creating a thin wrapper around it and switch to that but this will be at later stage (if at all).

Also note that Fiber uses fasthttp which doesn't  follow strictly the HTTP-spec and some things may not work as expected or entirely missing (eg. the last time I checked there wasn't support for http2, chunked body parsing, request/response body streaming, etc.).

>  replace fiber in places where will be much bigger traffic or it would not be significant change in performance so it is not worth it?

Don't rely on artificial benchmarks. In most cases the routers performance differences would be negliable as your app will be mostly dealing with db operations.
